### DISCUSSION TITLE
Search in GraphQL Documentation

### DISCUSSION THREAD

**Discussion by JacobT14** [Kong/insomnia#3728] [Ideas] [answered]
First off, I want to say THANK YOU for adding in schema documentation!  Huge win!  I would just like to open this ticket and say it would make it even better if you supported search (really just need to match on type, query, mutation, and subscription names).  That would be great!  Huge win on adding the schema though!

**Comment by welcome** [comment]
👋 Thanks for opening your first issue! If you're reporting a 🐞 bug, please make sure
you include steps to reproduce it. If you're requesting a feature 🎁, please provide real
use cases that would benefit. 👪

To help make this a smooth process, please be sure you have first read the
[contributing guidelines](https://github.com/getinsomnia/insomnia/blob/master/CONTRIBUTING.md).

**Comment by gschier** [comment]
Great suggestion, I totally forgot about search!

**Comment by narration-sd** [comment]
Just a reminder, since Search is kind of important. Schemas can get quite large, of course are very detailed, and also the names of actual Interfaces, for example, may not be as simple as expected, depending on GraphQL implementation.

A good search will find `CardsSite` when its interface for fragment is actually 'cards_cards_site', as a working example...

Insomnia is a very useful tool, and this will make it that much better. Especially for recommending it to help newcomers to GraphQL.

**Comment by narration-sd** [comment]
A p.s. - a quite good implementation you could look at is the explore or Playground in Gatsby. 

As you can see in the picture, Documentation and Schema are usually separate tabs or links. What you've implemented so far would be called Schema, and that list as mentioned can get very large, not something to be scanned with any ease. 

Thus, Documentation would be the interface that has Search. This way you can find what you need without needing to know up front the entire incantation for its name, and then you get also a nicely focused view, on just that item and its contents.

Cheers...

(here, the Documention/Schema panel has come in as an overlay from the right side, hidden to make maximum space for query and results, until you ask for it)

![gridsome-playground](https://user-images.githubusercontent.com/247945/65394425-c38eb380-dd42-11e9-9e37-299ef6e25df1.png)

**Comment by narration-sd** [comment]
Sorry, that example GraphiQL picture and description is actually from Gridsome :)

**Comment by asyba** [comment]
Yes please, add this!

**Comment by oheyjesse** [comment]
This would be awesome! It's available in GraphiQL & it is the main reason I not been able to switch over to insomnia :( Which sucks, as it's much better in many other ways! (The environment vars are awesome!)

**Comment by lipsumar** [comment]
Any idea if/when this will land ?

**Comment by mailopl** [comment]
One and a half year passed - any luck implementing this xD ?

**Comment by webberwang** [comment]
Without search it becomes nearly un-useable at scale

**Comment by lipsumar** [comment]
I switched to Altair, it's just unusable without search

**Comment by irimiab** [comment]
Yes, definitely a must-have!

**Comment by sebastian-fredriksson-bernholtz** [comment]
@wdawson Sorry for asking, but what's the reason the [ticket that this discussion came from](https://github.com/Kong/insomnia/issues/1620) was closed?

**Reply by reynolek** [reply]
Github does that automatically when converting an issue to a discussion. It preserves all conversation and moves it to the discussion layout.

**Comment by frizadiga** [comment]
any update on this feature request?

**Reply by wdawson** [reply]
We got a PR from @Hademumel #3388 🙌 Still needs some changes to get merged though.

**Comment by wdawson** [comment, marked as answer]
Good news everyone! This was [included in Insomnia 2021.6.0!](https://insomnia.rest/changelog#:~:text=GraphQL%20Schema%20Search) 🎉 

Happy searching
